sycl/doc/design/spirv-extensions/SPV_INTEL_joint_matrix.asciidoc

```diff
@@ -192,11 +200,15 @@ It must be the result of a constant instruction. +
  +
 'Scope' is memory scope for operations on the matrix. It must be the
 result of a constant instruction with scalar 'integer type'. +
+ +
+'Use' is an optional parameter that shows where in a math operation the matrix
+is used. It must be the result of a constant instruction with scalar
```

Comment:
[`Integer type`](https://registry.khronos.org/SPIR-V/specs/unified1/SPIRV.html#_types) is already defined as scalar in the core spec, so you can simplify this definition  and probably spell `integer` starting with uppercase letter to highlight that you are referring to an existing term.

Also: do we need to explicitly specify that the value is expected to be from `Use` table above and not just any arbitrary integer (like 42, for example)?

Reply:
well, it makes sense, but "'integer type' scalar" is a commonly used wording across https://registry.khronos.org/SPIR-V/specs/unified1/SPIRV.pdf . Regarding the values - added this info, though there is some misalignments in the spec, guess, which should be corrected.

Reply:
I think it would be best to document this similar to the way [Scope \<id\>](https://registry.khronos.org/SPIR-V/specs/unified1/SPIRV.html#Scope_-id-)s are documented:

> Must be an \<id\> of a 32-bit integer scalar. Its value is expected to be one of the values in the table below. If validation rules or the client API require a constant \<id\>, it is invalid for it to not be one of these values. If non-constant \<id\> are allowed, behavior is undefined if \<id\> is not one of these values.

Since the Use is an optional parameter what is the behavior if the optional parameter is omitted?

Reply:
Thanks for the great suggestion, Ben! I've also aligned Layout parameter with this definition. And I'll adopt it for Scope as well.

>> Since the Use is an optional parameter what is the behavior if the optional parameter is omitted?

Per new revision it's now a mandatory parameter, before that it was up to the devices' backends to decide.